File: ProjectAssignment-2/src/main/java/com/endava/projectassignment2/service/UserService.java
package com.endava.projectassignment2.service;

import com.endava.projectassignment2.dto.DeactivateUserDto;
import com.endava.projectassignment2.dto.NewPasswordRequestDto;
import com.endava.projectassignment2.dto.UserLoginDto;
import com.endava.projectassignment2.dto.UserPasswordResetDto;
import com.endava.projectassignment2.dto.UserRegisterDto;
import com.endava.projectassignment2.dto.UserResponseDto;
import com.endava.projectassignment2.dto.UserUpdateDto;
import com.endava.projectassignment2.util.Page;

import reactor.core.publisher.Mono;

public interface UserService {
	
	Mono<Page<UserResponseDto>> getAllUsers(int page , int size);
	
	Mono<Page<UserResponseDto>> search(String keyword ,int page , int size);
	
	Mono<UserResponseDto> getUserById(long id);
	
	Mono<UserResponseDto> loginUser(UserLoginDto userLoginDto); 
	
	Mono<UserResponseDto> registerUser(UserRegisterDto userRegisterDto);
	
	Mono<UserResponseDto> updateUser(UserUpdateDto userUpdateDto);
		
	Mono<Void> deactivateUser(DeactivateUserDto deactivateUserDto);
	
	Mono<Void> resetPassword(UserPasswordResetDto userPasswordResetDto);
	
	Mono<String> forgotPassword(String email);
	
	Mono<Void> setNewPassword(NewPasswordRequestDto newPasswordRequestDto);
	
	Mono<Boolean> checkTokenExists(String token);
	
	Mono<Void> deactivateUserById(long id);
	
	Mono<Void> activateUserById(long id);
	
	Mono<Void> resetLockByUserId(long id);
		
}

File: Frontend/src/services/userService.js
import { URLCONSTANTS } from "../constants/urlConstants";
import apiClient from "./apiClient"; 


export const getUsers = async () => {
  const response = await apiClient.get(URLCONSTANTS.USERS);
  return response;
};

export const searchUsers = async (keyword,page,size) => {
  const queryParams = new URLSearchParams();
  
  if (keyword) queryParams.append("keyword", keyword);
  if (page) queryParams.append("page", page);
  if (size) queryParams.append("size", size);

  const response = await apiClient.get(`${URLCONSTANTS.SEARCH_USERS}?${queryParams.toString()}`);
  return response;
};

export const getUserById = async (id) => {
  const response = await apiClient.get(`${URLCONSTANTS.USERS}/${id}`);
  return response;
};

export const registerUser = async (userRegisterDto) => {
  const response = await apiClient.post(`${URLCONSTANTS.USER_REGISTER}`,userRegisterDto);
  return response;
};

export const loginUser = async (userLoginDto) => {
  const response = await apiClient.post(`${URLCONSTANTS.USER_LOGIN}`,userLoginDto);
  return response;
};

export const forgotPassword = async (email) => {
  const response = await apiClient.put(`${URLCONSTANTS.FORGOT_PASSWORD}?email=${email}`);
  return response;
};

export const validateToken = async (token) => {
  const response = await apiClient.get(`${URLCONSTANTS.VALIDATE_TOKEN}?token=${token}`);
  return response;
};

export const setNewPassword = async (newPasswordRequestDto) => {
  const response = await apiClient.put(`${URLCONSTANTS.SET_NEW_PASSWORD}`,newPasswordRequestDto);
  return response;
};

export const updateUser = async (userUpdateDto) => {
  const response = await apiClient.put(`${URLCONSTANTS.USER_UPDATE}`,userUpdateDto);
  return response;
};


export const resetPassword = async (UserPasswordResetDto) => {
  const response = await apiClient.put(`${URLCONSTANTS.PASSWORD_RESET}`,UserPasswordResetDto);
  return response;
};

export const deactivateUser = async (deactivateUserDto) => {
  const response = await apiClient.put(`${URLCONSTANTS.DEACTIVATE}`,deactivateUserDto);
  return response;
};


